clippy_lints/src/unconditional_recursion.rs

```diff
@@ -23,8 +23,8 @@ declare_clippy_lint! {
     /// implementations.
     ///
     /// ### Why is this bad?
-    /// This is a hard to find infinite recursion that will crash any code
-    /// using it.
+    /// Infinite recursion in trait implementation can cause crashes
```

Comment:
Is `will` inaccurate? `can` implies it might not happen

Reply:
As pointed out in <https://github.com/rust-lang/rust-clippy/pull/14385#issuecomment-2710811876>, in presence of tail-call optimizations the infinite recursion might loop forever without ever crashing.

Reply:
I parsed it as "can (crash or hang)" which implies a third possibility

We can make it clear with something like "will either ..."

Reply:
Great, done